src/components/my-widgets-score-semester-storage.jsx

```diff
@@ -1,66 +1,212 @@
+				`Showing ${Math.min(tmpVals.length, state.startIndex + 1)} to ${Math.min(tmpVals.length, state.endIndex)} of 
+				${tmpVals.length} entries (filtered from ${Math.min(state.storageData?.length, MAX_ROWS)} total entries)`
+
+			setRowVals(tmpVals)
+			setState({...state, footerText: text})
+		}
+	}, [state.isLoading,
+		state.selectedTable,
+		state.startIndex,
+		state.endIndex,
+		state.storageData,
+		state.selectedTableName,
+		state.pages,
+		state.storageData,
+		searchInput
+	])
+
+	const onChangePageCount = useCallback(newValue => {
+		const numPages = Math.ceil(state.storageData?.length/newValue)
+		const pagesArr = Array.from({length: numPages}, (_, i) => i + 1) // Generates list of ints 1 to numPages
+		const startIndex = 0
+		const endIndex = Math.min(state.storageData?.length, newValue)
+		const _selectedTable = state.storageData?.slice(startIndex, endIndex)
+		setState({...state, rowsPerPage: newValue, selectedTable: _selectedTable, startIndex: startIndex, endIndex: endIndex, pageNumber: 0, pages: pagesArr})
+	}, [state])
+
+	const onChangeTable = useCallback(_selectedTableName => {
+		const numPages = Math.ceil(state.storageData?.length/state.rowsPerPage)
+		const pagesArr = Array.from({length: numPages}, (_, i) => i + 1) // Generates list of ints 1 to numPages
+		const startIndex = 0
+		const endIndex = Math.min (state.storageData?.length, state.rowsPerPage + 1)
+		const _selectedTable = state.storageData?.slice(startIndex, endIndex)
+		const _isTruncated = _selectedTable.length > MAX_ROWS
+		setState({...state, selectedTable: _selectedTable, selectedTableName: _selectedTableName, isTruncated: _isTruncated, startIndex: startIndex, endIndex: endIndex, pageNumber: 0, pages: pagesArr})
+	}, [state])
+
+	const onChangePageNumber = useCallback(newPageNum => {
+		const startIndex = Math.min(state.storageData?.length, newPageNum*state.rowsPerPage)
+		const endIndex = Math.min (state.storageData?.length, (newPageNum + 1)*state.rowsPerPage)
+		const _selectedTable = state.storageData?.slice(startIndex, endIndex)
+		setState({...state, startIndex: startIndex, endIndex: endIndex, selectedTable: _selectedTable, pageNumber: newPageNum})
+	}, [state])
+
+	const getPaginateBtns = useCallback(() => {
+		let pagesArr = state.pages?.map((val, index) => {
+			return(<a className={`paginate_button ${index === state.pageNumber ? 'current' : ''}`}
+				aria-controls="DataTables_Table_1"
+				tabIndex="0"
+				onClick={() => {onChangePageNumber(index)}}
+				key={index}>
+				{index+1}
+			</a>)
+		})
+
+		// Compresses the buttons
+		if (pagesArr.length > 7) {
+			const paginateIconL = <span className="ellipsis" key={999}>…</span>
+			const paginateIconR = <span className="ellipsis" key={888}>…</span>
+			const curPage = state.pageNumber + 1
+			const numPages = pagesArr.length
+			let paginateType = numPages - curPage >= 4 ?  
+				(curPage <= 4 ? 
+					"left" : 
+					"middle") :
+				"right"
+
+			switch(paginateType) {
+				case ("right"):
```

Comment:
prefer no parenthesis 
```suggestion
				case "right":
```